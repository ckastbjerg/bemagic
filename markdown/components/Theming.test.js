/* eslint-env jest */
const Theming = require('./Theming');

test('it throws', () => {
    expect(() => Theming()).toThrow();
});

test('it works', () => {
    global.config = { namespace: 'prefix', themeClass: 'theme' };

    expect(Theming({
        classes: 'block',
        tag: 'div',

    })).toMatchSnapshot();
});
