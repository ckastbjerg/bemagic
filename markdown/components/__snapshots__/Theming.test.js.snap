exports[`test it works 1`] = `
"
        
        <a name=Theming></a>
        #### Theming
    
        The \`block\` supports [\`theming\`](../CONVENTIONS.md) and can change its
        appearance when paired with the \`prefix-theme\` class.
        Note however, that themes may apply only for parts of the component.

        \`\`\`html
        <div class=\"block prefix-theme\">…</div>
        <!-- or using nesting -->
        <div class=\"prefix-theme\">
            <div class=\"block\">…</div>
        </div>
        \`\`\`
        <sub>See the [Component Overview](#component-overview) for more details.</sub>
    "
`;
